src/libLookup/Lookup.cpp

```diff
@@ -3174,7 +3174,14 @@ void Lookup::SenderTxnBatchThread() {
   }
   LOG_MARKER();
 
+  if (m_startedTxnBatchThread) {
+    LOG_GENERAL(WARNING,
+                "The last TxnBatchThread hasn't finished, discard this time");
+    return;
+  }
+
   auto main_func = [this]() mutable -> void {
+    m_startedTxnBatchThread = true;
```

Comment:
Do you think  better to use mutex?

Reply:
i have experienced piled up thread for sending txn batch which actually consumed all the cpu power, so it's better to discard it if too busy otherwise it may lead the untimely sending of txn packet

Reply:
yes, my suggestion is rather than boolean, use mutex

Reply:
use atomic bool.

Reply:
okay